cmd/release-controller/sync_release_payload.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/release-controller/pkg/apis/release/v1alpha1"
+	releasecontroller "github.com/openshift/release-controller/pkg/release-controller"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+	"sort"
+)
+
+func (c *Controller) processReleasePayloadEvent(obj interface{}) {
+	switch t := obj.(type) {
+	case *v1alpha1.ReleasePayload:
+		klog.Infof("Processing ReleasePayload Event for: %s/%s", t.Namespace, t.Name)
+		c.addReleasePayloadQueueKey(queueKey{namespace: t.Namespace, name: t.Name})
+	default:
+		utilruntime.HandleError(fmt.Errorf("couldn't get key for object %T", obj))
+	}
+}
+
+func (c *Controller) syncReleasePayload(key queueKey) error {
+	// Get the ReleasePayload resource with this namespace/name
+	payload, err := c.releasePayloadLister.ReleasePayloads(key.namespace).Get(key.name)
+	if err != nil {
+		// The ReleasePayload resource may no longer exist, in which case we stop processing.
+		if errors.IsNotFound(err) {
+			utilruntime.HandleError(fmt.Errorf("payload '%s/%s' in work queue no longer exists", key.namespace, key.name))
+			return nil
+		}
+		return err
+	}
+	klog.Infof("Syncing ReleasePayload: '%s/%s'", payload.Namespace, payload.Name)
+
+	// This is the entry point, for the release-controller, to respond to any updates to ReleasePayloads as they occur...
+
+	// TODO: Add business logic here
```

Comment:
what do you plan for this to eventually do?

Reply:
I added this with the expectation that the release-controller will probably may need to react to state changes of the ReleasePayloads.  In an ideal world, hopefully this can be removed after all is said and done! :crossed_fingers:

Reply:
Removed